src/tools/compiletest/src/runtest/run_make.rs

```diff
@@ -268,7 +259,17 @@ impl TestCx<'_> {
             p
         };
 
-        let mut rustc = Command::new(&self.config.rustc_path);
+        // run-make-support and run-make tests are compiled using the stage0 compiler
+        // If the stage is 0, then the compiler that we test (either bootstrap or an explicitly
+        // set compiler) is the one that actually compiled run-make-support.
+        let stage0_rustc = if self.config.stage == 0 {
+            self.config.rustc_path.clone()
+        } else {
+            let mut p = host_build_root.join("stage0").join("bin").join("rustc");
+            p.set_extension(env::consts::EXE_EXTENSION);
+            p
+        };
```

Comment:
This should unconditionally use config.rustc_path, right? Even when testing stage1, I would expect `stage0/bin/rustc` to not exist if `rustc` is set in `config.toml`.

Reply:
The `run-make-support` library is compiled with the bootstrap compiler (or `build.rustc`, if set). We need to compile `rmake.rs` with the exact same compiler. If we unconditionally used `config.rustc_path` here, it would defeat the point of this PR, which is trying to compile rms and rmake.rs using the stage0 compiler.

You are right that this won't work if `build.rustc` is configured and you want to test stage1. Not sure if that is ever useful? I thought that the use-case for `build.rustc` is to combine it with `--stage 0`, to actually test the passed compiler.

If we want to support this use-case, we'd need to pass the actual stage0 compiler from bootstrap, rather than guessing its location here.

Reply:
> I thought that the use-case for build.rustc is to combine it with --stage 0, to actually test the passed compiler.

Nope, it's major use case is for distros that want to use the bootstrap compiler they built themself rather than downloading an official release from the internet.

> The run-make-support library is compiled with the bootstrap compiler (or build.rustc, if set). We need to compile rmake.rs with the exact same compiler. If we unconditionally used config.rustc_path here, it would defeat the point of this PR, which is trying to compile rms and rmake.rs using the stage0 compiler.

I thought run-make-support would get compiled with the compiler at config.rustc_path too.

Reply:
Hm yeah. We probably need to make sure that we do build run-make-support with the configured stage 0 compiler (which would be `build.rustc`), but we can pass `RUSTC_BOOTSTRAP=-1` to force that to deny any unstable/nightly feature usages in run-make-support.

Note that this will always risk a divergence between the stage 0 compiler *we* use (typically a beta rustc) versus the stage 0 rustc distros want to use, but I think that's an acceptable risk and at worst distros could just delete/disable a few run-make tests.

Reply:
> Nope, it's major use case is for distros that want to use the bootstrap compiler they built themself rather than downloading an official release from the internet.

Sure, but do they actually run the tests with the custom bootstrap compiler? :thinking: Anyway, I will pass the path of the compiler from bootstrap, it is a more robust solution anyway.

> I thought run-make-support would get compiled with the compiler at config.rustc_path too.

No, that's what has happened before this PR, the goal of this PR is to always compile rms with stage0/build.rustc, to avoid needless rebuilds :)